include/irr/asset/CMTLPipelineMetadata.h

```diff
@@ -0,0 +1,99 @@
+#ifndef __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+#define __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+
+#include "irr/asset/IPipelineMetadata.h"
+
+namespace irr {
+namespace asset
+{
+
+class CMTLPipelineMetadata final : public IPipelineMetadata
+{
+public:
+    struct SMtl
+    {
+        std::string name;
+
+        //Ka
+        core::vector3df_SIMD ambient = core::vector3df_SIMD(1.f);
+        //Kd
+        core::vector3df_SIMD diffuse = core::vector3df_SIMD(1.f);
+        //Ks
+        core::vector3df_SIMD specular = core::vector3df_SIMD(1.f);
+        //Ke
+        core::vector3df_SIMD emissive = core::vector3df_SIMD(1.f);
+        //Tf
+        core::vector3df_SIMD transmissionFilter = core::vector3df_SIMD(1.f);
+        //Ns, specular exponent in phong model
+        float shininess = 32.f;
+        //d
+        float opacity = 1.f;
+        //illum
+        uint32_t illumModel = 0u;
+        //-bm
+        float bumpFactor = 1.f;
+
+        //PBR
+        //Ni, index of refraction
+        float IoR = 1.6f;
+        //Pr
+        float roughness = 0.f;
+        //Pm
+        float metallic = 0.f;
+        //Ps
+        float sheen;
+        //Pc
+        float clearcoatThickness;
+        //Pcr
+        float clearcoatRoughness;
+        //aniso
+        float anisotropy = 0.f;
+        //anisor
+        float anisoRotation = 0.f;
```

Comment:
same as above (I presume PBR and above are mutually exclusive)

Reply:
if mutually exclusive then the UBO struct could be a union.

Reply:
well, i think only `ambient` and `shininess` is of no use for PBR. Fun fact: all of these summed up is exactly 128 bytes